src/Listener/DefaultSender.php

```diff
@@ -192,4 +182,41 @@ public function onSend(SendBatchEvent $e) {
     $e->setCompleted($completed);
   }
 
+  /**
+   * Determine if an SMTP error is temporary or permanent.
+   *
+   * @param string $message
+   *   PEAR error message.
+   * @return bool
+   *   TRUE - Temporary/retriable error
+   *   FALSE - Permanent/non-retriable error
+   */
+  protected function isTemporaryError($message) {
+    // SMTP response code is buried in the message.
+    $code = preg_match('/ \(code: (.+), response: /', $message, $matches) ? $matches[1] : '';
+
+    if (strpos($message, 'Failed to write to socket') !== FALSE) {
+      return TRUE;
+    }
+
+    // Register 5xx SMTP response code (permanent failure) as bounce.
+    if ($code{0} === '5') {
```

Comment:
Looking at this again, I think there's a slight flaw here, $code could be an empty string and this $code{0} doesn't exist.